lib/tree.rb

```diff
@@ -61,7 +86,7 @@ def tree_data
       diameter_sum += apple.diameter
     end
 
-    avg_diameter = # It's up to you to calculate the average diameter for this harvest.
+    avg_diameter = basket.size > 0 ? diameter_sum*1.0/basket.size : 0
```

Comment:
This is overly complex. Remember that average is just (summation of parts / # of parts). Check out my PR if you need another hint.